core/consensus/authority/impl/authority_manager_impl.cpp

```diff
@@ -96,7 +97,65 @@ namespace kagome::authority {
       }
     }
 
-    {  // observe last finalized blocks
+    primitives::AuthorityList authorities;
+    {  // get voter set id at last finalized block
+      const auto &hash = finalized_block_hash;
+      auto header_res = block_tree_->getBlockHeader(hash);
+      if (header_res.has_error()) {
+        log_->critical("Can't get header of block {}: {}",
+                       hash,
+                       header_res.error().message());
+        return false;
+      }
+      const auto &header = header_res.value();
+
+      auto res = trie_storage_provider_->setToEphemeralAt(header.state_root);
+      BOOST_ASSERT_MSG(
+          res.has_value(),
+          "Must be set ephemeral bath on existing state root; deq");
```

Comment:
```suggestion
          "Setting ephemeral bath on an existing state root must succed; qed");
```
I wouldn't be so sure about it, some I/O error might occur and it's normal in runtime.

Reply:
Anyway, I don't think that authority module can argue about such things in storage module anyway, unless there's an explicit contract from the storage module.